index.js

```diff
@@ -159,21 +161,37 @@ class S3Adapter {
   getFileLocation(config, filename) {
     const fileName = filename.split('/').map(encodeURIComponent).join('/');
     if (this._directAccess) {
+      const fileKey = `${this._bucketPrefix}${fileName}`;
+
+      let presignedUrl = '';
+      if (this._presignedUrl) {
+        const params = { Bucket: this._bucket, Key: fileKey, Expires: this._presignedUrlExpires };
+        presignedUrl = this._s3Client.getSignedUrl('getObject', params);
+      }
       if (this._baseUrl) {
+        let directAccessFileKey = fileKey;
+        if (this._baseUrlDirect) {
+          directAccessFileKey = fileName;
```

Comment:
Can you please check if there is any way to simplify this whole code block. It seems to be overly complex like this variable overwriting.

Reply:
I made some changes and according to the lint the complexity remained the same (6).

> _I didn't leave the lint rule (`"complexity": ["error", 6]`) I used because other functions are even more complex!_